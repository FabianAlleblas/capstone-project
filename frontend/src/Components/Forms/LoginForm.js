import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import useForm from '../../hooks/useForm'
import Button from '../Buttons/Button'
import RegistrationModal from '../Modals/RegistrationModal'

LoginForm.propTypes = {
  userRegistration: PropTypes.func.isRequired,
  userLogin: PropTypes.func.isRequired,
}

export default function LoginForm({ userRegistration, userLogin }) {
  const {
    handleInputChange,
    formData,
    validatePassword,
    isPasswordValid,
  } = useForm()

  return (
    <Form onSubmit={handleLogin}>
      <Input
        name="email"
        type="email"
        placeholder="E-Mail"
        onChange={handleInputChange}
        required
      />
      <Input
        name="password"
        type="text"
        placeholder="Password"
        onChange={handleInputChange}
        required
      />
      {!isPasswordValid && (
        <WarningText>
          Password must contain at least 1 digit, 1 capital letter, 1 special
          character and to have a length of min. 8 characters
        </WarningText>
      )}
      <ButtonWrapper>
        <Button>Login</Button>
        <Button onClick={openRegistrationModal} secondaryStyle>
          Sign Up
        </Button>
      </ButtonWrapper>
      {/* <RegistrationModalFixed /> */}
    </Form>
  )

  function handleLogin(event) {
    event.preventDefault()
    userLogin(formData)
    event.target.password.blur()
  }

  function openRegistrationModal() {
    if (validatePassword()) {
    }
  }
}

const Form = styled.form`
  background-color: var(--primary-dark);
  display: grid;
  gap: 20px;
  place-items: center;
  width: 100%;
`

const Input = styled.input`
  border-radius: 22px;
  border: none;
  color: var(--form-font-color);
  font-family: 'Josefin sans', sans-serif;
  font-size: 1rem;
  padding: 14px 20px 10px;
  width: 100%;

  &:focus {
    box-shadow: 0 0 4px var(--shadow) inset;
    outline: none;
  }
`

const ButtonWrapper = styled.div`
  display: grid;
  gap: 20px;
  padding: 20px 0 0;
  width: 60%;
`

const WarningText = styled.p`
  color: var(--warning-color);
  font-size: 0.75rem;
  text-align: center;
`
const RegistrationModalFixed = styled(RegistrationModal)`
  height: 100vh;
  left: 0;
  position: absolute;
  top: 0;
`
