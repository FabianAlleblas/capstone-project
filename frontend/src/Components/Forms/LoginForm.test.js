import { render } from '@testing-library/react'
import user from '@testing-library/user-event'
import LoginForm from './LoginForm'

const onSubmitMock = jest.fn()

describe('AddPlantForm', () => {
  it('calls onSubmit with correct data, resets form and calls onSubmitMock', () => {
    const { getByPlaceholderText, getByText } = render(
      <LoginForm userRegistration={onSubmitMock} userLogin={onSubmitMock} />
    )

    user.type(getByPlaceholderText('E-Mail'), 'john@doe.com')
    user.type(getByPlaceholderText('Password'), '12345')

    user.click(getByText('Login'))

    expect(onSubmitMock).toHaveBeenCalledWith({
      email: 'john@doe.com',
      password: '12345',
    })

    expect(getByPlaceholderText('E-Mail')).toHaveValue('')
    expect(getByPlaceholderText('Password')).toHaveValue('')

    expect(onSubmitMock).toHaveBeenCalled()
  })
  it('calls onSubmitMock by clicking the sign up button', () => {
    const { getByText } = render(
      <LoginForm userRegistration={onSubmitMock} userLogin={onSubmitMock} />
    )

    user.click(getByText('Sign Up'))

    expect(onSubmitMock).toHaveBeenCalled()
  })
})
