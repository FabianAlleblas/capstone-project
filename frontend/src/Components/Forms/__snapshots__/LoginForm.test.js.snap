// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders correctly 1`] = `
<form
  class="LoginForm__Form-sc-5lwuot-0 kNnjuC"
>
  <input
    class="LoginForm__Input-sc-5lwuot-1 LMXKV"
    name="email"
    placeholder="E-Mail"
    required=""
    type="email"
  />
  <input
    class="LoginForm__Input-sc-5lwuot-1 LMXKV"
    name="password"
    placeholder="Password"
    required=""
    type="password"
  />
  <div
    class="LoginForm__ButtonWrapper-sc-5lwuot-2 fqactG"
  >
    <button
      class="Button__PrimaryButton-sc-6pjoi6-0 pAbUY"
    >
      Login
    </button>
    <button
      class="Button__SecondaryButton-sc-6pjoi6-1 cSPa-Dk"
      type="button"
    >
      Sign Up
    </button>
  </div>
</form>
`;
